plugins/source/googleanalytics/client/fetch.go

```diff
@@ -0,0 +1,59 @@
+package client
+
+import (
+	"context"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	analyticsdata "google.golang.org/api/analyticsdata/v1beta"
+)
+
+func fetch(tableName string, request *analyticsdata.RunReportRequest) schema.TableResolver {
+	return func(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+		c := meta.(*Client)
+		logger := c.Logger().With().Str("table", tableName).Logger()
+
+		req := c.service.Properties.RunReport(c.PropertyID, request).Context(ctx)
+
+		dates, err := genDates(ctx, c, tableName)
+		if err != nil {
+			return err
+		}
+
+		for date := range dates {
+			dateStr := date.Format(layout)
+			request.DateRanges = []*analyticsdata.DateRange{{StartDate: dateStr, EndDate: dateStr}}
+
+			var fetched int64
+			var gotOther bool
+			for {
+				request.Offset = fetched
+
+				resp, err := req.Do()
+				if err != nil {
+					return err
+				}
+
+				res <- convertRows(resp, date)
+
+				fetched += int64(len(resp.Rows))
+
+				if fetched >= resp.RowCount {
+					break
+				}
+
+				gotOther = gotOther || resp.Metadata.DataLossFromOtherRow
+			}
+
+			if gotOther {
+				logger.Warn().Str("date", dateStr).Msg("got (other) row, consider modifying report")
+			}
+
+			// We save current date here, even with data loss (as the report should be edited to get all the data)
```

Comment:
Could you clarify this comment? I'm assuming the "data loss" here is referring to if the report excluded some past days. In that case I'd suggest rephrasing this as "We save current date here, even if past dates were excluded by filters"

Reply:
`Data loss` refers to `(other)` grouped value present (see [this](https://support.google.com/analytics/answer/13331684) doc).

Reply:
Ah okay, let's add this context to the comment then

Reply:
(We should also add it to the user-facing docs somewhere, so you don't need to read the code to find it)

Reply:
Done in a3281ab